### DISCUSSION TITLE
Sorry Friday Noob Time help need i have issue with push to my repo

### DISCUSSION THREAD

**Discussion by thorrrr** [twpayne/chezmoi#2886] [Q&A] [answered]
I used
git remote add origin https://github.com/redacted/dotfiles.git
git branch -M main
git push -u origin main

But my terminal says NO see below i have no idea why it errors i followed the docs.
But  i guess it must be user error 

![image](https://user-images.githubusercontent.com/8170673/227491341-f12a56b0-d96e-43f3-bd12-314f9f8ab2bf.png)

**Comment by halostatue** [comment]
Your `dotfiles` repo on GitHub probably doesn’t have a `main` branch, but a `master` branch. Instead of `git branch -M main`, use `git branch -m main` before pushing.

When I searched Stack Overflow for `refspec main does not match any`, I found [an exact match question](https://stackoverflow.com/questions/70179719/git-error-src-refspec-main-does-not-match-any) that has this an other possible answers.

**Reply by thorrrr** [reply]
That is not the case mine is called main in Github Desktop
For some reason the 

![image](https://user-images.githubusercontent.com/8170673/227555031-d4b93da5-ac9f-4616-a539-ed67cb547f39.png)

Also i cannot get rid of the code screen 

![image](https://user-images.githubusercontent.com/8170673/227555541-f577e4da-5bfb-451c-989d-e5c0a113e56f.png)

**Comment by thorrrr** [comment]
I have got the following setup to test 

![image](https://user-images.githubusercontent.com/8170673/227556061-46f6f46f-790d-425f-9220-d4c22ec11b48.png)


![image](https://user-images.githubusercontent.com/8170673/227556678-b594eac5-9752-4aa2-b24d-d64124006970.png)

**Reply by halostatue** [reply]
There are some additional suggestions in the Stack Overflow question which may be worth trying. Beyond that, I don’t know what to suggest to resolve this as it’s a git issue, not a chezmoi issue.

If your dotfiles are a new repo entirely, then this can be bypassed with:

1. Delete the remote repo in GitHub and recreate it. When you do so, tell GitHub put a README and perhaps a `.gitignore` in it.
2. Clone the repo to your local system and move your test files into it.
3. `git add . && git commit -m "Test commit" && git push`

At that point, you should be off to the races.

**Comment by bradenhilton** [comment]
What is the output of

```console
$ git branch -vva
```

?

**Comment by thorrrr** [comment]
[dale@dalelaptop]~% $ git branch -vva
zsh: $: command not found...

**Comment by twpayne** [comment, marked as answer]
This is a git question, not a chezmoi question. Please look for your solution elsewhere.
